### PR TITLE
TRUNK-5666 : Add tests to UserByNameComparator

### PR DESCRIPTION
**Pull request by ayush2297**
Added 2 tests for compare function to add coverage to UserByNameComparator
- the tests handle the part where one of the two inputs in null 

- [ x ] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

- [ x ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

- [ x ] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

- [ x ] All new and existing **tests passed**.

- [ x ] My pull request is **based on the latest changes** of the master branch.

### CODE REVIEW

**Comment by gitcliff** [comment]
@ayush2297 have you seen the travis ci failure ?
Also you can have a look at this https://wiki.openmrs.org/display/docs/Pull+Request+Tips

**Comment by ayush2297** [comment]
there was a minor error in license header.. fixed it .

File: api/src/test/java/org/openmrs/util/UserByNameComparatorTest.java [hunk 1]
```diff
@@ -9,58 +9,110 @@
...
 	 */
 	@Test
 	public void compare_shouldSortUsersByPersonNames() {
-		
-		Person person1 = new Person();
-		person1.addName(new PersonName("givenName", "middleName", "familyName"));
-		User user1 = new User(person1);
-		Person person2 = new Person();
-		person2.addName(new PersonName("givenName", "middleNamf", "familyName"));
-		User user2 = new User(person2);
```
Comment (gitcliff) [inline api/src/test/java/org/openmrs/util/UserByNameComparatorTest.java:41]: @ayush2297 are these changes part of the ticket or formatting changes ?

**Comment by gitcliff** [comment]
@ayush2297  add ticket ID to the commit

**Comment by HerbertYiga** [comment]
@ayush2297,well done on this just cant find the ticket ID on the description so as to know what was needed for this

File: api/src/test/java/org/openmrs/util/UserByNameComparatorTest.java (hunk 1 above)
Comment (ayush2297) [reply api/src/test/java/org/openmrs/util/UserByNameComparatorTest.java:41]: these are formatting changes.. i didn't want to repeat code so i moved these declarations at test level and set them up in setup() method and then reused the variables in my tests.

Comment (gitcliff) [reply api/src/test/java/org/openmrs/util/UserByNameComparatorTest.java:41]: oh okay

**Comment by ayush2297** [comment]
> add ticket ID to the commit

added ticket ID to the commit message.

File: api/src/test/java/org/openmrs/util/UserByNameComparatorTest.java [hunk 2]
```diff
@@ -9,58 +9,110 @@
...
+		// added the users randomly
+		listToSort.add(nullUser);
+		listToSort.add(user1);
+
+		listToSort.sort(new UserByNameComparator());
+		Iterator<User> sortedListIterator = listToSort.iterator();
+
+		Assert.assertTrue("Expected user1 to be the first in the sorted user list but wasn't",
+			user1.equals(sortedListIterator.next()));
+	}
```
Comment (Am-Coder) [inline api/src/test/java/org/openmrs/util/UserByNameComparatorTest.java:117]: What if there is a list having multiple null users? Testing this for a list having just 2 users drops this case. A more robust test can make use of atleast 2 users and 2 null users.

Comment (ayush2297) [reply api/src/test/java/org/openmrs/util/UserByNameComparatorTest.java:117]: okay... makes sense.. i'll try it.. 👍 
let me know if there are more such things that need to be worked on..  :)

Comment (ayush2297) [reply api/src/test/java/org/openmrs/util/UserByNameComparatorTest.java:137]: done.. please check once..

Comment (ayush2297) [reply api/src/test/java/org/openmrs/util/UserByNameComparatorTest.java:137]: i wanted to make sure there's no error when it comes to null-null (consecutive) comparison..... but now I realised, that condition would get tested somewhere in the alternate null test too.. :/ 
sorry for the confusion.. should i remove the consecutive one and push the code? 
is there anything else to be covered?

Comment (Amanmishra-gep) [reply api/src/test/java/org/openmrs/util/UserByNameComparatorTest.java:137]: Yes, remove that test.

Comment (Am-Coder) [reply api/src/test/java/org/openmrs/util/UserByNameComparatorTest.java:137]: Yes remove the test

**Comment by gitcliff** [comment]
@ayush2297 kindly have a look at the merge conflicts and the travis failure

**Comment by Wandji69** [comment]
@ayush2297 have any progress on the Travis failure??

**Comment by Wandji69** [comment]
@dkayiwa, the ticket for this PR has been closed I think this PR should be close as well.

**Comment by dkayiwa** [comment]
Thanks @Wandji69 for following it up.
